client/src/hooks/useRequestResourceForm.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import { useUser } from 'hooks/useUser'
+import api from 'api'
+
+export default (resource = {}, requestId) => {
+  const { user, token } = useUser()
+  const [request, setRequest] = React.useState({ material: resource.id })
+
+  React.useEffect(() => {
+    if (requestId && !request.id) fetchRequest(requestId)
+  })
+
+  const setAttribute = (attribute, value) => {
+    const updatedRequest = { ...request }
+    updatedRequest[attribute] = value
+    setRequest(updatedRequest)
+  }
+
+  const getAttribute = (attribute) => {
+    return request[attribute]
+  }
+
+  const setAddressAttribute = (attribute, value) => {
+    const updatedRequest = { ...request }
+    const address = updatedRequest.address || {}
+
+    address[attribute] = value
+
+    updatedRequest.address = address
+
+    setRequest(updatedRequest)
+  }
+
+  const getAddressAttribute = (attribute) => {
+    const address = request.address || {}
+    return address[attribute]
+  }
+
+  const fetchRequest = async () => {
+    const materialRequestRequest = await api.requests.get(requestId, token)
+    if (materialRequestRequest.isOk) setRequest(materialRequestRequest.response)
+  }
+
+  const createResourceRequest = async () => {
+    const saveRequest = { ...request }
+
+    if (saveRequest.address) {
+      const { address } = saveRequest
+
+      // was already saved
+      if (address.id) {
+        delete address.id
+        address.save_for_reuse = false
+      }
+
+      const addressRequest = await api.user.addresses.create(
+        user.id,
+        address,
+        token
+      )
+
+      if (addressRequest.isOk) {
+        saveRequest.address = addressRequest.response.id
+      } else {
+        console.log(addressRequest.response)
```

Comment:
Should we have another method for keeping track of this as we're going to set the linter to flag console.logs soon? Or is this ok for now?

Reply:
I need to leave this in here because reporting the error wont be enough, will need to figure out a way to just submit the address if this fails.